### ISSUE TITLE
Fixture apps with both `app` and `config` crash tests

### ISSUE THREAD

**Issue by danmcclain** [tomdale/ember-cli-addon-tests#2]
For some reason, a fixture that has both an `app` directory and a `config` directory causes AddonTestApp to fail and crash any test that occurs after it. I'm trying to capture more useful logging right now for this issue

Fail log for the first test failure:
```
     Error: the object {
  "code": 1
  "errors": [
    "Could not find module `dummy/app` imported from `~fastboot/app-factory`\n"
    "Error: Could not find module `dummy/app` imported from `~fastboot/app-factory`\n    at missingModule (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11625-7345
1-17f3zyy/under-test/host/fastboot-dist/assets/bower_components/loader.js/loader.js:135:1)\n    at findModule (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11625-
73451-17f3zyy/under-test/host/fastboot-dist/assets/bower_components/loader.js/loader.js:150:1)\n    at Module.reify (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-
11625-73451-17f3zyy/under-test/host/fastboot-dist/assets/bower_components/loader.js/loader.js:86:1)\n    at Module.build (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn
/T/d-11625-73451-17f3zyy/under-test/host/fastboot-dist/assets/bower_components/loader.js/loader.js:104:1)\n    at findModule (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c800
00gn/T/d-11625-73451-17f3zyy/under-test/host/fastboot-dist/assets/bower_components/loader.js/loader.js:152:1)\n    at Object.requireModule (/private/var/folders/v_/cq8q2tb52ldg
4fp30k9638c80000gn/T/d-11625-73451-17f3zyy/under-test/host/fastboot-dist/assets/bower_components/loader.js/loader.js:139:1)\n"
  ]
  "output": [
    "version: 1.13.15\n"
    "Building"
    "Building."
    "Building.."
    "Built project successfully. Stored in \"fastboot-dist\".\n"
    "Installing FastBoot npm dependencies\n"
  ]
  "signal": [null]
```

All the tests after that fail for
```
TypeError: Path must be a string. Received null
```

The branch exhibiting this behavior is here: https://github.com/danmcclain/ember-cli-fastboot/tree/add-host-whitelist-to-package-json

**Comment by kategengler** [comment]
I am also seeing these test failures on a fork without any changes https://travis-ci.org/kategengler/ember-cli-fastboot/builds/114067915

**Comment by danmcclain** [comment]
@kategengler I've traced your issue down to https://github.com/tomdale/ember-cli-addon-tests/blob/master/lib/utilities/pristine.js#L156-L171

TL;DR: Adds ember-data canary to `bower.json`, which is broken.

Once ember-data to an older version (and also updating the version of ember-cli to 2.3.0) in testing, I come across a new issue (in both cases):

```
  "errors": [
    "ENOENT: no such file or directory, lstat '/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/under-test/cookies/tmp/funnel-input_base_path-gFvsXYJY.tmp/0/index.html'\n"
    "Error: ENOENT: no such file or directory, lstat '/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/under-test/cookies/tmp/funnel-input_base_path-gFvsXYJY.tmp/0/index.html'\n    at Error (native)\n    at Object.fs.lstatSync (fs.js:887:18)\n    at symlink (/private/var/folders/v_/cq8
q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_modules/ember-cli/node_modules/symlink-or-copy/index.js:63:26)\n    at Function.symlinkOrCopySync [as sync] (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_modules/ember-cli/node_modules/symlink-or-copy/i
ndex.js:58:5)\n    at Funnel._copy (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_modules/ember-cli/node_modules/broccoli-funnel/index.js:398:19)\n    at Funnel.processFile (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_mo
dules/ember-cli/node_modules/broccoli-funnel/index.js:381:8)\n    at Funnel.applyPatch [as _applyPatch] (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_modules/ember-cli/node_modules/broccoli-funnel/index.js:298:12)\n    at Funnel.<anonymous> (/private/var/folders/v_/cq
8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_modules/ember-cli/node_modules/broccoli-funnel/index.js:250:10)\n    at Array.forEach (native)\n    at Funnel.processFilters (/private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11628-14325-1ohpjg3/pristine/node_modules/ember-cli/node_modules
/broccoli-funnel/index.js:249:9)\n"
```

**Comment by danmcclain** [comment]
I ran `ember init` on the result of an addon, seems like files are getting removed accidentally from the pristine app:

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        app/app.js
        app/index.html
        app/templates/application.hbs
```

I have verified that not all the files from the pristine app are getting copied correctly:

```
╭─dan@Tron  /private/var/folders/v_/cq8q2tb52ldg4fp30k9638c80000gn/T/d-11629-43663-6ltijc/under-test/cookies ‹2.2.0›
╰─$ ls -al app
total 8
drwxr-xr-x   7 dan  cloakproxy  238 Mar  9 22:17 .
drwxr-xr-x  21 dan  cloakproxy  714 Mar  9 22:17 ..
drwxr-xr-x   3 dan  cloakproxy  102 Mar  9 22:17 controllers
drwxr-xr-x   3 dan  cloakproxy  102 Mar  9 22:17 instance-initializers
-rw-r--r--   1 dan  cloakproxy  225 Mar  9 22:17 router.js
drwxr-xr-x   3 dan  cloakproxy  102 Mar  9 22:17 routes
drwxr-xr-x   3 dan  cloakproxy  102 Mar  9 22:17 templates
```

I'm going to continue digging into this

**Comment by danmcclain** [comment]
My issue was actually my own misconfiguration, but I did find an issue with `cpr` because of all the digging I did, plus there is a broken canary build of ember-data available via bower
